build.rs

```diff
@@ -49,7 +49,7 @@ fn main() -> Result<()> {
     let target = env::var("TARGET").expect("TARGET was not set");
     let mut base_config = cc::Build::new();
 
-    language_std(&mut base_config, "c++17");
+    language_std(&mut base_config, "c++2a");
```

Comment:
The wonderful thing about standards is that there's so many of them
```suggestion
    // try to match the standards expected by the compilers in GitHub Actions images, as of October 2021
    if target.contains("windows") {
        language_std(&mut base_config, "c++20");
    } else {
        language_std(&mut base_config, "c++17");
    }
```